From 87b048392e9b9b85dd435fe44a4bbf7ef8e2f9a6 Mon Sep 17 00:00:00 2001
Message-Id: <87b048392e9b9b85dd435fe44a4bbf7ef8e2f9a6.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Hong Liu <hong.liu@intel.com>
Date: Thu, 6 Nov 2014 14:26:45 +0800
Subject: [PATCH 29/34] FOR_UPSTREAM [VPG]: drm/i915: avoid NULL pointer
 deference to sprite_unpin_work

Currently we just set old sprite_unpin_work to NULL if it is not
finished when new flip comes.

This can causes NULL pointer deference issue in irq handler.
And also we lose references to fb in the unpin_work.

Issue: GMINL-3685
Change-Id: Iaf7c5a1aafcd6acd6ebdf81d09d500484ade41b7
Signed-off-by: Hong Liu <hong.liu@intel.com>
---
 drivers/gpu/drm/i915/intel_sprite.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index ed2c819..9e848d8 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -1667,7 +1667,6 @@ intel_update_plane(struct drm_plane *plane, struct drm_crtc *crtc,
 			spin_unlock_irqrestore(&dev->event_lock, flags);
 			kfree(work);
 			drm_vblank_put(dev, intel_crtc->pipe);
-			intel_crtc->sprite_unpin_work = NULL;
 			DRM_ERROR("flip queue: crtc already busy\n");
 			return -EBUSY;
 		}
-- 
1.7.9.5

